docker/test/integration/minifi/core/DockerTestCluster.py

```diff
@@ -318,3 +318,25 @@ def wait_for_metric_class_on_prometheus(self, metric_class, timeout_seconds):
 
     def wait_for_processor_metric_on_prometheus(self, metric_class, timeout_seconds, processor_name):
         return PrometheusChecker().wait_for_processor_metric_on_prometheus(metric_class, timeout_seconds, processor_name)
+
+    def wait_for_peak_memory_usage(self, minimum_peak_memory_usage: int, timeout_seconds: int) -> bool:
+        start_time = time.perf_counter()
+        current_peak_memory_usage = get_peak_memory_usage(get_minifi_pid())
+        while (time.perf_counter() - start_time) < timeout_seconds:
+            if current_peak_memory_usage > minimum_peak_memory_usage:
+                return True
+            current_peak_memory_usage = get_peak_memory_usage(get_minifi_pid())
+            time.sleep(1)
+        print(f"Peak memory usage ({current_peak_memory_usage}) didnt exceed minimum asserted peak memory usage {minimum_peak_memory_usage}")
```

Comment:
I think we shold use the `logging` library instead, same for `wait_for_maximum_memory_usage`

Reply:
Good idea, I've changed it in https://github.com/apache/nifi-minifi-cpp/commit/c7b837f8f6bd5150c2e1de5c0ecf210df9a98225#diff-4487753f519234543da32b7926ab6cfc45c79ae2467eca71c8f6c269c7195eefR333